File: src/db/fennec/api/grpc/client/error/FennecException.java
package db.fennec.api.grpc.client.error;

public class FennecException extends Exception {

    private int code;
    private String message;
    private Throwable internal;

    public FennecException(int code, String message) {
        this(code, message, null);
    }

    public FennecException(int code, String message, Throwable internal) {
        this.code = code;
        this.message = message;
        this.internal = internal;
    }
}

File: src/db/fennec/error/FennecException.kt
package db.fennec.error

import java.lang.Exception

open class FennecException(val status: Status, final override val message: String, throwable: Throwable? = null)
    : Exception("${status.getDescriptor()} - $message", throwable) {

    val structuredMsg: String = "${status.getDescriptor()} - $message"
}